src/components/menu/menu-item/menu-item.component.js

```diff
@@ -38,40 +38,85 @@ const MenuItem = ({
   onSubmenuClose,
   overrideColor,
   rel,
-  isFocused,
   ...rest
 }) => {
-  const menuContext = useContext(MenuContext);
+  const {
+    inFullscreenView,
+    registerItem,
+    unregisterItem,
+    focusId,
+    menuType,
+    openSubmenuId,
+  } = useContext(MenuContext);
+  const menuItemId = useRef(guid());
   const submenuContext = useContext(SubmenuContext);
+  const {
+    registerItem: registerSubmenuItem,
+    unregisterItem: unregisterSubmenuItem,
+    submenuFocusId,
+    updateFocusId: updateSubmenuFocusId,
+    handleKeyDown: handleSubmenuKeyDown,
+    shiftTabPressed,
+  } = submenuContext;
   const ref = useRef(null);
-  const focusFromMenu = isFocused;
-  const focusFromSubmenu = submenuContext.isFocused;
-  const isChildSearch = useRef(false);
-  const childRef = useRef();
-  const { inFullscreenView } = menuContext;
-  const childrenItems = React.Children.map(children, (child) => {
-    if (child?.type === Search) {
-      isChildSearch.current = true;
+  const focusFromMenu = focusId === menuItemId.current;
+  const focusFromSubmenu = submenuFocusId
+    ? submenuFocusId === menuItemId.current
+    : undefined;
+  const searchRef = useRef(null);
+  const inputIcon = useRef(null);
+
+  inputIcon.current = ref.current?.querySelector(
+    "[data-element='input-icon-toggle']"
+  );
+  searchRef.current = ref.current?.querySelector("[data-element='input']");
```

Comment:
I'm surprised we don't use a specific data-element/data-component value for `Search` - but it appears not. This makes me nervous because in theory there might be other input elements inside a submenu, even though that's I agree unlikely. (Actually if there are we probably still want to be able to focus them - which makes the name a bit misleading imo.)

Reply:
This was renamed based off your previous comment

Reply:
I'll go with `inputRef` as you make a valid point